plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentModule.java

```diff
@@ -104,10 +109,24 @@ public void configure(Binder binder)
         public void configure(Binder binder)
         {
             MapBinder<String, RowEncoderFactory> encoderFactoriesByName = encoderFactory(binder);
-            encoderFactoriesByName.addBinding(AvroRowEncoder.NAME).toInstance((session, dataSchema, columnHandles) -> {
-                throw new TrinoException(NOT_SUPPORTED, "Insert not supported");
-            });
+            encoderFactoriesByName.addBinding(AvroRowEncoder.NAME).to(ConfluentAvroEncoderFactory.class).in(Scopes.SINGLETON);
             binder.bind(DispatchingRowEncoderFactory.class).in(SINGLETON);
         }
     }
+
+    private static class ConfluentAvroEncoderFactory
+            implements RowEncoderFactory
+    {
+        @Override
+        public RowEncoder create(ConnectorSession session, Optional<String> dataSchema, List<EncoderColumnHandle> columnHandles)
+        {
+            throw new TrinoException(NOT_SUPPORTED, "Insert not supported");
+        }
+
+        @Override
+        public boolean supportsMissingColumns()
```

Comment:
Why do the Confluent Avro encoder and the normal Avro encoder differ?

Reply:
Confluent's Avro encoder has a few extra bytes like a schema id.. so it differs in that aspect. So the writes are not supported for Confluent Schema Registry based tables.

Reply:
In that case I would suggest making the `supportsMissingColumns` to `true` since with how it is right now it looks more like the override is added because Confluent encoder doesn't support default values.